frontend/public/components/sidebars/resource-sidebar.jsx

```diff
@@ -85,22 +89,52 @@ const ResourceSamples = ({ samples, kindObj, downloadSampleYaml, loadSampleYaml
   />
 );
 
+const ResourceSnippets = ({ snippets, kindObj, insertSnippetYaml }) => (
+  <ResourceSidebarSnippets
+    snippets={snippets}
+    kindObj={kindObj}
+    insertSnippetYaml={insertSnippetYaml}
+  />
+);
+
 const ResourceSidebarContent = (props) => {
   const {
     downloadSampleYaml,
     height,
     isCreateMode,
     kindObj,
     loadSampleYaml,
+    insertSnippetYaml,
     yamlSamplesList,
   } = props;
   if (!kindObj) {
     return null;
   }
 
   const { label } = kindObj;
-  const samples = getResourceSidebarSamples(kindObj, yamlSamplesList);
+  const { samples, snippets } = getResourceSidebarSamples(kindObj, yamlSamplesList);
   const showSamples = !_.isEmpty(samples) && isCreateMode;
+  const showSnippets = snippets.length > 0;
+
+  const tabs = [];
+  if (showSamples) {
+    // TODO: Pre-determine if we have a schema
+    // Possible Related Bug: https://jira.coreos.com/browse/CONSOLE-1611
+    tabs.push({
```

Comment:
multiple elements can be pushed in one single push statement.
```
tabs.push({
      name: 'Schema',
      component: ResourceSchema,
    }, {
      name: 'Samples',
      component: ResourceSamples,
    });
```

Reply:
Yup - I am aware, but I don't feel that is the right approach. Right now Schema is tied to Samples and that's not strictly true. One can have samples without a schema (such as Pipelines).

I left it this way intentionally as a means to flag for removal later.